src/js/actions/guides.js

```diff
@@ -0,0 +1,313 @@
+    deleteGuide.writes = [locks.JS_DOC];
+    deleteGuide.transfers = [resetGuidePolicies];
+
+    var queryCurrentGuides = function (document) {
+        var docRef = documentLib.referenceBy.id(document.id);
+
+        return _getGuidesForDocumentRef(docRef)
+            .bind(this)
+            .then(function (guides) {
+                var payload = {
+                    document: document,
+                    guides: guides
+                };
+
+                return this.dispatch(events.document.GUIDES_UPDATED, payload);
+            });
+    };
+    queryCurrentGuides.reads = [locks.PS_DOC];
+    queryCurrentGuides.writes = [locks.JS_DOC];
+
+    // Event handlers for guides
+    var _guideSetHandler = null,
+        _guideDeleteHandler = null;
+
+    /**
+     * Register event listeners for guide edit/delete events
+     * 
+     * @return {Promise}
+     */
+    var beforeStartup = function () {
+        // Listen for guide set events
+        _guideSetHandler = function (event) {
+            var target = objUtil.getPath(event, "null._ref");
+
+            if (target && _.isArray(target) && target.length === 2 &&
+                target[0]._ref === "document" && target[1]._ref === "good") {
+                var payload = {
+                    documentID: target[0]._id,
+                    index: target[1]._index - 1, // PS indices guides starting at 1
+                    orientation: event.orientation._value,
+                    position: event.position._value
+                };
+
+                this.flux.actions.guides.setGuide(payload);
+            }
+        }.bind(this);
+        descriptor.addListener("set", _guideSetHandler);
+
+        // Listen for guide delete events
+        _guideDeleteHandler = function (event) {
+            var target = objUtil.getPath(event, "null._ref");
+
+            // Mind the reversal of references compared to "set"
+            if (target && _.isArray(target) && target.length === 2 &&
+                target[1]._ref === "document" && target[0]._ref === "good") {
+                var payload = {
+                    documentID: target[1]._id,
+                    index: target[0]._index - 1 // PS indices guides starting at 1
+                };
+
+                this.flux.actions.guides.deleteGuide(payload);
+            }
+        }.bind(this);
+        descriptor.addListener("delete", _guideDeleteHandler);
+
+        return Promise.resolve();
+    };
+    beforeStartup.modal = true;
+    beforeStartup.reads = [];
+    beforeStartup.writes = [];
+    beforeStartup.transfers = [];
+    
+    /**
+     * Remove event handlers.
+     *
+     * @return {Promise}
+     */
+    var onReset = function () {
+        descriptor.removeListener("set", _guideSetHandler);
+        descriptor.removeListener("delete", _guideDeleteHandler);
+
+        return Promise.resolve();
+    };
+    onReset.reads = [];
+    onReset.writes = [];
+
+    exports._getGuidesForDocumentRef = _getGuidesForDocumentRef;
```

Comment:
this is just a "maybe for a future revision" comment: Something rubs me the wrong way about exporting this "private" method.  Gut says there's a better way to factor this.